src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target.hidden === false) {
```

Comment:
```suggestion
    if (target.hidden === false) {
```
As task description says:

> Carnivore has a bite method, which takes a herbivore object and decreases the object's health by 50. The method does not work if it is another сarnivore, or the herbivore is currently hiding.

So you should also check if the target is an herbivore

Reply:
Да, но у плотоядных животных нет свойства hidden, потому проверяя одно, я исключаю другое, зачем еще проверка?

Reply:
Good point 🤔

Reply:
But for better reading and understanding it would be better to specify that check too